execution/execution_test.go

```diff
@@ -71,3 +72,27 @@ func TestExecutionHash(t *testing.T) {
 
 	require.NoError(t, quick.Check(f, nil))
 }
+
+func TestGetDuration(t *testing.T) {
```

Comment:
All time manipulation should use the time package.
We should never do stuff like `1000*1000*1000`, this is error prone.